course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -1,70 +1,115 @@
-                dc.courseInstance = dc.getFormattedCourseInstance(courseResult.data, membersResult.data)
+            if (response.data.course_instance_id == dc.courseInstanceId) {
+                courseInstances.instances[response.data.course_instance_id] = response.data;
+                // if people/members are fetched first, we don't want to
+                // overwrite the members attribute of dc.courseInstance
+                $.extend(dc.courseInstance,
+                    dc.getFormattedCourseInstance(response.data));
+                // TLT-2376: only sandbox and ILE courses are currently editable
+                var rc = response.data.course.registrar_code;
+                dc.editable = dc.isCourseInstanceEditable(rc);
+                dc.resetForm();
             } else {
                 $log.error(' CourseInstance record mismatch for id :'
-                    + dc.courseInstanceId + ',  fetched record for :' + courseResult.data.id);
+                    + dc.courseInstanceId + ',  fetched record for :'
+                    + response.data.id);
             }
         };
 
-        dc.setCourseInstance = function (id) {
+        dc.handlePeopleResponse = function(response) {
+            if (!dc.hasOwnProperty('courseInstance')) { }
+            dc.courseInstance['members'] = response.data.count;
+        };
+
+        dc.fetchCourseInstanceDetails = function (id) {
 
-            var course_url = djangoUrl.reverse(
-                'icommons_rest_api_proxy',
-                ['api/course/v2/course_instances/' + id + '/']);
+            var course_url = djangoUrl.reverse(dc.apiProxy,
+                [dc.apiBase + id + '/']);
 
-            var members_url = djangoUrl.reverse(
-                'icommons_rest_api_proxy',
-                ['api/course/v2/course_instances/'
-                + id + '/people/']);
+            var members_url = djangoUrl.reverse(dc.apiProxy,
+                [dc.apiBase + id + '/people/']);
 
-            var coursePromise = $http.get(course_url)
-                .error(dc.handleAjaxError);
+            $http.get(course_url)
+                .then(dc.handleCourseInstanceResponse,
+                    function cleanUpFailedCourseDetailsGet(response) {
+                        dc.handleAjaxErrorResponse(response);
+                        dc.resetGlobalAlerts();
+                        dc.alerts.global.fetchCourseInstanceFailed = {
+                            show: true,
+                            details: response.statusText || 'None'};
+                });
 
-            var membersPromise = $http.get(members_url)
-                .error(dc.handleAjaxError);
+            $http.get(members_url)
+                .then(dc.handlePeopleResponse,
+                    function cleanUpFailedCourseMembersGet(response) {
+                        dc.handleAjaxErrorResponse(response);
+                        dc.alerts.form.fetchMembersFailed = {
```

Comment:
The fetch members failed message is a 'form'-level message -- it appears next to the People label, not at the top of the form. So we don't want to/need to clear global messages, and don't want to scroll (although we could try to scroll to this particular error message, if there's some way of doing it).

<img width="1012" alt="screen shot 2016-03-22 at 10 40 44 am" src="https://cloud.githubusercontent.com/assets/8975317/13955372/b9a35402-f01a-11e5-8bf0-891b60836d9e.png">

Probably merits more comments at the very least!